### PR TITLE
Issue 81 - Deleting an object closely followed by publishing an objec…

### PR DESCRIPTION
**Pull request by LiilyZhang**
…t fails to delete first object

Signed-off-by: zhangl <zhangl@us.ibm.com>

### CODE REVIEW

File: swagger.json [hunk 1]
```diff
@@ -1834,6 +1834,12 @@
             "description": "Fetch the objects with expiration time before specified timestamp in RFC3339 format",
             "name": "expirationTimeBefore",
             "in": "query"
+          },
```
Comment (dabooz) [inline swagger.json:1837]: I think we need to update the ./docs/mms_swagger in anax repo

File: core/base/apiServer.go [hunk 2]
```diff
@@ -865,6 +865,11 @@ func handleObjectRequest(orgID string, objectType string, objectID string, write
 //   description: Fetch the objects with expiration time before specified timestamp in RFC3339 format
 //   required: false
 //   type: string
+// - name: deleted
+//   in: query
+//   description: Fetch the object with deleted marked to true
+//   required: false
+//   type: boolean
```
Comment (dabooz) [inline core/base/apiServer.go:872]: This has 3 values, it's not a boolean. if this is a boolean, the API caller will think that no value on the URL will mean false.

**Review by dabooz — CHANGES_REQUESTED** [review]

File: core/base/apiServer.go (hunk 2 above)
Comment (LiilyZhang) [reply core/base/apiServer.go:872]: @dabooz Should I change type to "string"? And in the description, say something like: if specified, value should be "true" or "false". If not specified, result will not be filtered by this field

Comment (dabooz) [reply core/base/apiServer.go]: yes, exactly

File: core/base/apiServer.go [hunk 3]
```diff
@@ -346,6 +346,7 @@ func handleShutdown(writer http.ResponseWriter, request *http.Request) {
 	setResponseHeaders(writer)
 
 	code, _, _ := security.Authenticate(request)
+	fmt.Printf("auth code : %d\n", code)
```
Comment (dabooz) [inline core/base/apiServer.go:349]: Is this a left over debug statement?

**Review by dabooz — CHANGES_REQUESTED** [review]

**Review by dabooz — APPROVED** [review]

**Pull request merged**